src/components/providers/dataProvider.ts

```diff
@@ -0,0 +1,207 @@
+    lawfulBasis: "Consent",
+    agreementEvent: "Opt-in",
+    timestamp: "2023-08-26T18:52:33z",
+  },
+];
+
+const fakeViewLogsDataProvider = [
+  {
+    id: 0,
+    action: "admin@retail.com logged in",
+    category: "Security",
+    timestamp: "2023-09-22 05:26:58 +0000 UTC",
+  },
+  {
+    id: 1,
+    action: "admin@retail.com logged in",
+    category: "Security",
+    timestamp: "2023-09-22 05:26:58 +0000 UTC",
+  },
+  {
+    id: 2,
+    action: "admin@retail.com logged in",
+    category: "Security",
+    timestamp: "2023-09-22 05:26:58 +0000 UTC",
+  },
+  {
+    id: 3,
+    action: "admin@retail.com logged in",
+    category: "Security",
+    timestamp: "2023-09-22 05:26:58 +0000 UTC",
+  },
+  {
+    id: 4,
+    action: "admin@retail.com logged in",
+    category: "Security",
+    timestamp: "2023-09-22 05:26:58 +0000 UTC",
+  },
+];
+
+const fakeWebhookDataProvider = [
+  { id: 0, callBackURL: "https://webhook.igrant.io", status: "Active" },
+  { id: 1, callBackURL: "https://webhook.igrant.io", status: "Active" },
+  { id: 2, callBackURL: "https://webhook.igrant.io", status: "Active" },
+  { id: 3, callBackURL: "https://webhook.igrant.io", status: "Active" },
+  { id: 4, callBackURL: "https://webhook.igrant.io", status: "Active" },
+];
+
+export const fakeConsentBBDataProvider = fakeDataProvider({
+  dataagreement: [
+    {
+      id: 0,
+      Name: "User Registration",
+      Version: "1.0.0",
+      MethodOfUse: "Data Using Service",
+      PublishFlag: "Saved",
+      LawfulUsage: "Contractual",
+    },
+    {
+      id: 1,
+      Name: "User Registration",
+      Version: "1.1.0",
+      MethodOfUse: "Data Source",
+      PublishFlag: "Saved",
+      LawfulUsage: "Contractual",
+    },
+    {
+      id: 2,
+      Name: "User Registration",
+      Version: "1.0.2",
+      MethodOfUse: "Data Using Service",
+      PublishFlag: "Saved",
+      LawfulUsage: "Consent",
+    },
+    {
+      id: 3,
+      Name: "User Registration",
+      Version: "1.0.0",
+      MethodOfUse: "Data Using Service",
+      PublishFlag: "Saved",
+      LawfulUsage: "Contractual",
+    },
+    {
+      id: 4,
+      Name: "User Registration",
+      Version: "1.1.0",
+      MethodOfUse: "Data Source",
+      PublishFlag: "Saved",
+      LawfulUsage: "Contractual",
+    },
+    {
+      id: 5,
+      Name: "User Registration",
+      Version: "1.0.2",
+      MethodOfUse: "Data Using Service",
+      PublishFlag: "Saved",
+      LawfulUsage: "Consent",
+    },
+  ],
+  personaldata: fakePersonalDataDataProvider,
+  userrecords: fakeUserRecordDataProvider,
+  viewlogs: fakeViewLogsDataProvider,
+  webhooks: fakeWebhookDataProvider,
+});
+
+const dataAgreementDataProvider = {
+  getList: (resource: any, params: any) => {
+    return HttpService.getDataAgreement()
+      .then((dataAgreements) => {
+        let page_size = params.pagination.perPage;
+        let page_number = params.pagination.page;
+        const PaginateddataAgreements = paginate(
```

Comment:
Non-constant variables names should be `camelCase` - `paginatedDataAgreements`. Can we consistently follow this ?